ipa-core/src/protocol/ipa_prf/malicious_security/verifier.rs

```diff
@@ -145,14 +132,15 @@ where
 
     // generate & evaluate recursive streams
     // to compute last array
-    let mut stream: Pin<Box<dyn Stream<Item = F> + Send>> = Box::pin(
-        recurse_u_or_v::<_, _, λ_FIRST>(u_or_v_stream, &table_first),
-    );
+    let mut iterator: Box<dyn Iterator<Item = F>> = Box::new(recurse_u_or_v::<_, _, λ_FIRST>(
+        u_or_v_iterator,
+        &table_first,
+    ));
     // all following recursion except last one
     for lagrange_table in tables.iter().take(recursions_after_first - 1) {
-        stream = Box::pin(recurse_u_or_v::<_, _, λ>(stream, lagrange_table));
+        iterator = Box::new(recurse_u_or_v::<_, _, λ>(iterator, lagrange_table));
```

Comment:
why boxing it?

Reply:
If you can manage to remove the Box, that would be even better.

Reply:
the dyn type always needs a Box since the size is not known at compile time. I need a dyn type since I apply lambda times the recurse_u_or_v on the input steam, i.e. recurse_u_or_v(...recurse_u_or_v(u_or_v_iterator)...) and every intermediate output type of recurse is a different type of iterator even though they are all iterators with Items of type F.